### PR TITLE
Move normalization step into the dispatch flow

### PR DESCRIPTION
**Pull request by bkrmendy**
https://github.com/user-attachments/assets/d184b155-e18a-42d6-ba84-1532f6ebd280

## Problem
The tailwind style plugin can't differentiate between a prop being removed from the inline style prop, and that prop never having been there at all. For example, when a flex gap is removed by dragging it in the negative direction, it goes from positive -> 0 -> being removed from the style prop. When the normalisation step kicks in, there's no gap prop in the inline style, so the tailwind plugin doesn't do anything. To make matters worse, when the gap property is removed while the interaction is in progress, the value coming from the generated tailwind class takes precedence, so the flex gap jumps back to its original size.

## Fix
The problem of removing props in the normalization step is fixed by
1. moving the normalization step in the display flow
2. making the normalization step aware that some props need to be removed
3. find out which elements to normalize/which props to remove from the dispatched actions and the strategies, and running the normalization on those elements and prop.

The problem of prevent the CSS classes to kick in when a prop is removed (for example, when the inline `gap` prop is removed in the flex gap strategy), this PR adds a postprocessing step alongside the normalization step, that patches an allowlisted set of props to a "zero" value, which simulates the prop not being there at all

### Details
- added a test for removing the gap prop set from Tailwind
- added a `propertiesToRemove` param to `normalizeFromInlineStyle`, and updated the tailwind style plugin to use that prop to remove classes from the `className` prop
- added the code that finds out which elements/props to normalize from the dispatched actions
- added the code that finds out which elements/props to normalize from the strategies, and that tells which elements to patch during an interaction

**Manual Tests:**
I hereby swear that:

- [x] I opened a hydrogen project and it loaded
- [x] I could navigate to various routes in Play mode

Fixes #[ticket_number] (<< pls delete this line if it's not relevant)

### CODE REVIEW

**Review by balazsbajorics — APPROVED** [review]

**Review by gbalint — APPROVED** [review]

File: editor/src/components/canvas/plugins/tailwind-style-plugin.ts [hunk 1]
```diff
@@ -59,6 +60,55 @@ function getTailwindClassMapping(classes: string[], config: Config | null) {
...
+    return mapDropNulls(
+      ([key, value]) => (typeof key !== 'object' ? null : { key, value }),
+      Object.entries(styleProp.value),
+    )
+  }
+
+  return null
+}
+
+function getRemoveUpdates(properties: PropertyPath[]) {
```
Comment (Rheeseyb) [inline editor/src/components/canvas/plugins/tailwind-style-plugin.ts:88]: Some of these functions are missing return types. Can you add them please

Comment (bkrmendy) [reply editor/src/components/canvas/plugins/tailwind-style-plugin.ts]: added them in https://github.com/concrete-utopia/utopia/pull/6549/commits/c84c9525c010928f343789d907b974835e0fdb9c

**Review by Rheeseyb — APPROVED** [review]

File: editor/src/components/canvas/plugins/style-plugins.ts [hunk 2]
```diff
@@ -23,7 +26,7 @@ export const Plugins = {
 } as const
 
 export function getActivePlugin(editorState: EditorState): StylePlugin {
-  if (isTailwindEnabled()) {
+  if (isFeatureEnabled('Tailwind') && isTailwindEnabled()) {
```
Comment (seanparsons) [inline editor/src/components/canvas/plugins/style-plugins.ts:29]: Should that feature switch be inside `isTailwindEnabled`?

Comment (bkrmendy) [reply editor/src/components/canvas/plugins/style-plugins.ts:29]: good point, I put the feature flag check here to be explicit about the fact that this won't kick in until the FS is removed

File: editor/src/components/editor/actions/action-utils.ts [hunk 3]
```diff
@@ -362,3 +364,74 @@ export function simpleStringifyActions(
     .map((a) => simpleStringifyAction(a, indentation))
     .join(`,\n${spacing}`)}\n${spacingBeforeClose}]`
 }
+
+export function getElementsToNormalizeFromCommands(commands: CanvasCommand[]): ElementPath[] {
+  return mapDropNulls((command) => {
+    switch (command.type) {
```
Comment (seanparsons) [inline editor/src/components/editor/actions/action-utils.ts:370]: Minor, but this being exhaustive ensures we have to fix it when adding a new `CanvasCommand`.

**Review by seanparsons — APPROVED** [review]

**Pull request merged**